polygeist/tools/cgeist/Test/Verification/sycl/functions.cpp

```diff
@@ -101,28 +101,404 @@ SYCL_EXTERNAL void range_size(sycl::range<2> r) {
   keep(r.size());
 }
 
-// CHECK-MLIR: func.func @_Z8method_1N4sycl3_V14itemILi2ELb1EEE(%arg0: !sycl_item_2_1_)
-// CHECK-MLIR-SAME: attributes {llvm.cconv = #llvm.cconv<spir_funccc>, llvm.linkage = #llvm.linkage<external>, passthrough = ["norecurse", "nounwind", "convergent", "mustprogress"]} {
-// CHECK-MLIR-NEXT: %c0_i32 = arith.constant 0 : i32
-// CHECK-MLIR-NEXT: %0 = memref.alloca() : memref<1x!sycl_item_2_1_>
-// CHECK-MLIR-NEXT: affine.store %arg0, %0[0] : memref<1x!sycl_item_2_1_>
-// CHECK-MLIR-NEXT: %1 = "polygeist.memref2pointer"(%0) : (memref<1x!sycl_item_2_1_>) -> !llvm.ptr<!sycl_item_2_1_>
-// CHECK-MLIR-NEXT: %2 = llvm.addrspacecast %1 : !llvm.ptr<!sycl_item_2_1_> to !llvm.ptr<!sycl_item_2_1_, 4>
-// CHECK-MLIR-NEXT: %3 = "polygeist.pointer2memref"(%2) : (!llvm.ptr<!sycl_item_2_1_, 4>) -> memref<?x!sycl_item_2_1_, 4>
-// CHECK-MLIR-NEXT: %4 = sycl.call(%3, %c0_i32) {Function = @get_id, MangledName = @_ZNK4sycl3_V14itemILi2ELb1EE6get_idEi, Type = @item} : (memref<?x!sycl_item_2_1_, 4>, i32) -> i64
-// CHECK-MLIR-NEXT: return
-// CHECK-MLIR-NEXT: }
+// CHECK-MLIR-LABEL: func.func @_Z8id_get_0N4sycl3_V12idILi1EEEi(%arg0: !sycl_id_1_, %arg1: i32) attributes {llvm.cconv = #llvm.cconv<spir_funccc>, llvm.linkage = #llvm.linkage<external>, passthrough = ["norecurse", "nounwind", "convergent", "mustprogress"]} {
```

Comment:
This is good, it shows that the FE generates MLIR containing the new operation. Can you also add a test (a new RUN step) that verifies these new operations are lowered to a sycl.call ? I think is good to verify in the lit tests that we convert all the new operation to sycl.call (at least currently).

Reply:
We cannot test that currently. I'll add a test for that in the new PR which introduces a new pass with this intermediate step.